tests/sentry/snuba/metrics/test_metrics_layer/test_metrics_enhanced_performance.py

```diff
@@ -50,6 +62,229 @@ def test_valid_filter_include_meta_derived_metrics(self):
             key=lambda elem: elem["name"],
         )
 
+    def test_alias_on_different_metrics_expression(self):
+        now = before_now(minutes=0)
+
+        for v_transaction, count in (("/foo", 1), ("/bar", 3), ("/baz", 2)):
+            for value in [123.4] * count:
+                self.store_metric(
+                    org_id=self.organization.id,
+                    project_id=self.project.id,
+                    type="distribution",
+                    name=TransactionMRI.MEASUREMENTS_LCP.value,
+                    tags={"transaction": v_transaction, "measurement_rating": "poor"},
+                    timestamp=int(time.time()),
+                    value=value,
+                    use_case_id=UseCaseKey.PERFORMANCE,
+                )
+
+        metrics_query = MetricsQuery(
+            org_id=self.organization.id,
+            project_ids=[self.project.id],
+            select=[
+                MetricField(
+                    op="count",
+                    metric_name=TransactionMetricKey.MEASUREMENTS_LCP.value,
+                    alias="count_lcp",
```

Comment:
If somebody mix alias for column in select and order by? E.g.
```
select=[
                MetricField(
                    op="count",
                    metric_name=TransactionMetricKey.MEASUREMENTS_LCP.value,
                    alias="count_lcp",
...
orderby=[
                OrderBy(
                    MetricField(
                        op="count",
                        metric_name=TransactionMetricKey.MEASUREMENTS_LCP.value,
                        alias="foo_bar_count_lcp",
                    ),
                    Direction.DESC,
                ),
```
it's just an idea for negative test case.

Reply:
I know we check that column from select should be in orderby. But what about aliases?
JFYI it's valid query in click-house:
```
d69e28edac27 :) select project_id as ttt from outcomes_hourly_local order by project_id as tttt;

SELECT project_id AS ttt
FROM outcomes_hourly_local
ORDER BY project_id AS tttt ASC

Query id: bb94eed4-2a0d-4a61-a8c4-59417c357d93

┌─ttt─┐
│   2 │
└─────┘

1 rows in set. Elapsed: 0.004 sec.
```

Reply:
but this query is invalid:
```
SELECT project_id AS ttt
FROM outcomes_hourly_local
ORDER BY tttt ASC

Received exception from server (version 21.6.1):

Code: 47. DB::Exception: Received from localhost:9000. DB::Exception: Missing columns: 'tttt' while processing query: 'SELECT project_id AS ttt FROM outcomes_hourly_local ORDER BY tttt ASC', required columns: 'project_id' 'tttt', maybe you meant: ['project_id'].
```

Reply:
> If somebody mix alias for column in select and order by? E.g.

I delegate this behaviour to snuba, but it should not matter right? because the ordering behaviour would be the same regardless of the alias applied to it

> I know we check that column from select should be in orderby. But what about aliases?
JFYI it's valid query in click-house:

Query 1 is the format we produce so I think it should be fine

Reply:
agree.